RecoTracker/TkDetLayers/src/Phase2EndcapSingleRing.cc

```diff
@@ -0,0 +1,161 @@
+#include "Phase2EndcapSingleRing.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "TrackingTools/DetLayers/interface/DetLayerException.h"
+#include "TrackingTools/DetLayers/interface/MeasurementEstimator.h"
+#include "TrackingTools/GeomPropagators/interface/HelixForwardPlaneCrossing.h"
+#include "TrackingTools/DetLayers/interface/rangesIntersect.h"
+#include "TrackingTools/DetLayers/interface/ForwardRingDiskBuilderFromDet.h"
+
+#include "LayerCrossingSide.h"
+#include "DetGroupMerger.h"
+#include "CompatibleDetToGroupAdder.h"
+
+#include "TkDetUtil.h"
+#include "DataFormats/GeometryVector/interface/VectorUtil.h"
+
+using namespace std;
+
+typedef GeometricSearchDet::DetWithState DetWithState;
+
+class DetGroupElementZLess {
+public:
+  bool operator()(DetGroup a, DetGroup b) {
+    return (std::abs(a.front().det()->position().z()) < std::abs(b.front().det()->position().z()));
+  }
+};
```

Comment:
This already exists in a similar form in Phase2EndcapRing. Would it make sense to unify?